### ISSUE TITLE
Creating "Azure Cache for Redis" using Azure Native

### ISSUE THREAD

**Issue by johnkors** [pulumi/pulumi-azure-native#641]
One cannot create a valid "Azure Cache for Redis" resource using the Azure Native Provider

## Expected behavior
An Azure Cache for Redis resource.

## Current behavior
Fails with a "Invalid Request" and message saying that "Family" is missing.

## Steps to reproduce

1. Use Azure Native
2. Try to setup a Azure Cache for Redis:

```
      var redis = new AzureNative.Cache.Redis("redis", new AzureNative.Cache.RedisArgs
        {
            Sku = new AzureNative.Cache.Inputs.SkuArgs
            {
                Capacity = 1,
                Name = "Premium",
                // Family = "C" // does not compile, cannot set family.
            }

        });
```

## Context (Environment)
Azure Native

## Affected feature
Azure Native & https://www.pulumi.com/docs/reference/pkg/azure-native/cache/redis/

**Comment by johnkors** [comment]
Workaround: Use Azure Classic provider. It let's you set Family: https://www.pulumi.com/docs/reference/pkg/azure/redis/cache/

```
var exampleCache = new Azure.Redis.Cache("exampleCache", new Azure.Redis.CacheArgs
{
    Family = "C"
});
```

**Comment by mikhailshilkov** [comment]
This is really interesting, thank you for reporting. A better workaround for now is to use an explicit version:

```cs
var redis = new Pulumi.AzureNative.Cache.V20200601.Redis("redis", new RedisArgs
        {
            Sku = new Pulumi.AzureNative.Cache.V20200601.Inputs.SkuArgs
            {
                Capacity = 1,
                Name = "Premium",
                Family = "C"
            }
        });
```

which does compile.

I'll get back to you when I know why our top-level resource is broken.

**Comment by johnkors** [comment]
Could you also clarify how to get a the `PrimaryConnectionString` when using Azure Native? It's missing compared to the Classic.

**Comment by mikhailshilkov** [comment]
Azure API doesn't return sensitive data by default, so you have to query the key explicitly with [`listrediskeys`](https://www.pulumi.com/docs/reference/pkg/azure-native/cache/listrediskeys/). Similarly to [this example](https://github.com/pulumi/templates/blob/master/azure-csharp/MyStack.cs#L33) (but for Redis).

**Comment by johnkors** [comment]
Ok, and then build the connectionString from access keys in code. Thanks.

**Comment by mikhailshilkov** [comment]
> I'll get back to you when I know why our top-level resource is broken.

So, the reason is that both Redis and RedisEnterprise definitions live in the same resource provider Microsoft.Cache, use the same names for the SKU definition, but actually define different types:

- Redis Enterprise: https://github.com/Azure/azure-rest-api-specs/blob/d3f2ecf27be1e1043427f581f5b41687b722b04f/specification/redisenterprise/resource-manager/Microsoft.Cache/stable/2021-03-01/redisenterprise.json#L1128
- Redis: https://github.com/Azure/azure-rest-api-specs/blob/d3f2ecf27be1e1043427f581f5b41687b722b04f/specification/redis/resource-manager/Microsoft.Cache/stable/2020-06-01/redis.json#L1778

When we squash them both to the top level, the types clash. I'll need to think of a prevention mechanism.

**Comment by johnkors** [comment]
If they are different, why aren't they different types?

```csharp
new RedisEnterprise();
new Redis();
```

**Comment by mikhailshilkov** [comment]
The resources are different types but they both refer to a `sku` property of type `Sku`. Those types are actually different but we fail to detect this difference and think it's the same `Sku`.

TL;DR: we should fix this and make them different types.
